src/twisted/internet/defer.py

```diff
@@ -1661,8 +1663,8 @@ def _inlineCallbacks(
                 result = current_context.run(gen.send, result)
         except StopIteration as e:
             # fell off the end, or "return" statement
-            status.deferred.callback(getattr(e, "value", None))
-            return
+            should_callback = True
+            callback_val = getattr(e, "value", None)
```

Comment:
I think this is always present from 3.3+

```suggestion
            callback_val = e.value
```

Reply:
Asyncio also sets the result in the except block https://github.com/python/cpython/blob/3.9/Lib/asyncio/tasks.py#L265